spec/index.bs

```diff
@@ -114,33 +168,73 @@ interface GPUAdapter : GPUObjectBase {
 };
 </script>
 
+{{GPUAdapter}} has the following attributes:
+
+<dl dfn-type=attribute dfn-for=GPUAdapter>
+    : <dfn>\[[adapter]]</dfn>, of type [=adapter=], readonly
+    ::
+        An internal slot holding the [=adapter=] to which this {{GPUAdapter}}
+        refers.
+
+    : <dfn>name</dfn>
+    ::
+        A human-readable name identifying the adapter.
+        The contents are implementation-defined.
+
+    : <dfn>extensions</dfn>
+    ::
+        A {{GPUExtensions}} object which enumerates the extensions supported
+        by the user agent, and whether each extension is supported by the
+        underlying implementation.
+          - If an extension is not supported by the user agent,
+            it will not be present in the object.
+          - If an extension is supported by the user agent, but
+            not by the [=adapter=], it will be `false`.
+          - If an extension is supported by the user agent and
+            by the [=adapter=], it will be `true`.
+</dl>
+
 ### Getting an Adapter ### {#adapter-creation}
 
-<script type=idl>
-[Exposed=Window]
-interface GPU {
-    // May reject with DOMException  // TODO: DOMException("OperationError")?
-    Promise<GPUAdapter> requestAdapter(optional GPURequestAdapterOptions options);
-};
-</script>
+To get a {{GPUAdapter}}, use {{GPU/requestAdapter()}}.
 
 <script type=idl>
 dictionary GPURequestAdapterOptions {
-    GPUPowerPreference? powerPreference;
+    GPUPowerPreference powerPreference;
```

Comment:
We should explain what powerPreference should do to the device selection process, e.g.

```
- A preference of <dfn enum-value for="GPUPowerPreference">low-power</dfn> indicates a request for, as an example, the integrated GPU on a laptop with switchable graphics.
- A preference of <dfn enum-value for="GPUPowerPreference">high-performance</dfn> indicates a request for, as an example, the dedicated GPU on a laptop with switchable graphics, or a powerful external GPU.

Note: {{GPUPowerPreference}} serves only as a hint to the user agent. Depending on the exact hardware configuration, such as attached displays or removable GPUs, 
the device selected by the user agent may differ from the one intended by the provided {{GPURequestAdapterOptions}}. Ideally, the logic used to select the underlying
device will result in the same device being selected across multiple browsing sessions, given the same hardware configuration.
```

I don't think my definitions for the preferences are good enough, though.